### PR TITLE
Pass SSL properties from Glue Connection to MySQL

### PR DESCRIPTION
**Pull request by kukushking**
*Issue [#554](https://github.com/awslabs/aws-data-wrangler/issues/554):*

*Description of changes:*
Pass SSL properties from Glue Connection to PyMySQL. ~~Depends on [this PyMySQL PR](https://github.com/PyMySQL/PyMySQL/pull/980) merged & released first. ~~

UPD: In conversation with PyMySQL contributors it seems it's possible to pass `SSLContext` to PyMySQL so extra argument is not required anymore.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by jaidisido** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: GitHubCodeBuild8756EF16-sDRE8Pq0duHT
* Commit ID: 735ad12df22f42d50713a9168739e79dd8bb0dd8
* Result: FAILED
* [Build Logs](https://ozufyid077.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=03369573-9c5e-4a4d-b81f-a2f045205e9f%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by jaidisido** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: GitHubCodeBuild8756EF16-sDRE8Pq0duHT
* Commit ID: 35706a3494fb0a4188d76061ef268a1d3d1ba40c
* Result: FAILED
* [Build Logs](https://ozufyid077.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=5e0dfc9c-b3e1-4e31-9192-7838c0eebe83%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by jaidisido** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: GitHubCodeBuild8756EF16-sDRE8Pq0duHT
* Commit ID: cd1954075bc724a8d7c6387bce1620b82c900220
* Result: FAILED
* [Build Logs](https://ozufyid077.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=6053a39c-813a-4e14-8723-6796ad8b8c33%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by kukushking** [comment]
@jaidisido CB failed here because a new `aws-data-wrangler-mysql-ssl` connection that I added in this PR isn't available in the test infra yet. How do we usually go about it? Do I need to deploy manually?

**Comment by jaidisido** [comment]
Ideally, we would want to make this change permanent if it will be needed for future tests. So I would add/modify the cloudformation [template](https://github.com/awslabs/aws-data-wrangler/blob/2da313473e6426128fff0111ae819fb55ccc87ee/cloudformation/databases.yaml#L363) holding the connections first (you can change it as part of this PR). I would then update the stack in the AWS account we are using for testing

**Comment by kukushking** [comment]
That modification is already there (see diff). All, right, I'll update it in the account then.

**Comment by jaidisido** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: GitHubCodeBuild8756EF16-sDRE8Pq0duHT
* Commit ID: cd1954075bc724a8d7c6387bce1620b82c900220
* Result: FAILED
* [Build Logs](https://ozufyid077.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=4f13bf14-659d-4362-849f-48be9fc13aa3%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by jaidisido** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: GitHubCodeBuild8756EF16-sDRE8Pq0duHT
* Commit ID: cd1954075bc724a8d7c6387bce1620b82c900220
* Result: FAILED
* [Build Logs](https://ozufyid077.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=c37e5799-f14b-4387-b2cf-c5a51a6f9a11%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by jaidisido** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: GitHubCodeBuild8756EF16-sDRE8Pq0duHT
* Commit ID: cd1954075bc724a8d7c6387bce1620b82c900220
* Result: SUCCEEDED
* [Build Logs](https://ozufyid077.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=2330b691-a8ba-450b-ace2-a4020428753e%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

File: awswrangler/mysql.py [hunk 1]
```diff
@@ -139,6 +139,7 @@ def connect(
         read_timeout=read_timeout,
         write_timeout=write_timeout,
         connect_timeout=connect_timeout,
+        ssl=attrs.ssl_context,
```
Comment (jaidisido) [inline awswrangler/mysql.py:142]: Minor - I would put this line right after the `host=attrs.host` (line 137) just to keep all attributes arguments together

File: awswrangler/_databases.py [hunk 2]
```diff
@@ -71,6 +88,7 @@ def _get_connection_attributes_from_secrets_manager(
         host=secret_value["host"],
         port=secret_value["port"],
         database=_dbname,
+        ssl_context=None,
```
Comment (jaidisido) [inline awswrangler/_databases.py:91]: I guess there is really no way to specify/pass the ca data via secrets manager... Do you think we should clarify the docs in that regard?

File: cloudformation/databases.yaml [hunk 3]
```diff
@@ -384,6 +384,32 @@ Resources:
           PASSWORD:
             Ref: DatabasesPassword
         Name: aws-data-wrangler-mysql
+  MysqlGlueConnectionSSL:
+    Type: AWS::Glue::Connection
+    Properties:
+      CatalogId:
+        Ref: AWS::AccountId
+      ConnectionInput:
+        Description: Connect to Aurora (MySQL).
```
Comment (jaidisido) [inline cloudformation/databases.yaml:393]: Minor - "Connect to Aurora (MySQL) SSL enabled."

File: tests/test_mysql.py [hunk 4]
```diff
@@ -24,6 +24,10 @@ def test_connection():
     wr.mysql.connect("aws-data-wrangler-mysql", connect_timeout=10).close()
 
 
+def test_connection_ssl():
+    wr.mysql.connect("aws-data-wrangler-mysql-ssl", connect_timeout=10).close()
```
Comment (jaidisido) [inline tests/test_mysql.py:28]: Do you think we should be not only testing the ssl connection, but also use that connection for some of the mysql test queries? Was wondering if we could adapt the current pytest [fixture](https://github.com/awslabs/aws-data-wrangler/blob/2da313473e6426128fff0111ae819fb55ccc87ee/tests/test_mysql.py#L16) to take ssl enabled as an argument.

Something along the lines of:
```

@pytest.fixture(scope="function")
def mysql_con(ssl_enabled: bool = False):
    con_str = "aws-data-wrangler-mysql-ssl" if ssl_enabled else "aws-data-wrangler-mysql"
    con = wr.mysql.connect(con_str)
    yield con
    con.close()
```

**Review by jaidisido — COMMENTED** [review]
Looking great, thank you. I have left minor comments

File: awswrangler/_databases.py (hunk 2 above)
Comment (kukushking) [reply awswrangler/_databases.py:91]: Doesn't hurt to clarify, of course.

File: tests/test_mysql.py (hunk 4 above)
Comment (jaidisido) [reply tests/test_mysql.py:28]: Agreed, it's an overkill. One or two tests should be enough!

**Comment by jaidisido** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: GitHubCodeBuild8756EF16-sDRE8Pq0duHT
* Commit ID: 0dcccbc6c9cf4244f372e717e5b9c2e31571c074
* Result: SUCCEEDED
* [Build Logs](https://ozufyid077.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=a82e4127-2e63-4da2-9a63-587c0ff8e77e%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Pull request merged**